src/python/grpcio/grpc/_cython/_cygrpc/completion_queue.pyx

```diff
@@ -74,7 +68,41 @@ cdef class CompletionQueue:
     with self.poll_condition:
       self.is_polling = False
       self.poll_condition.notify()
+    return self.interpret_event(event)
+
+  def pluck(self, records.OperationTag tag, records.Timespec deadline=None):
+    # Plucking a 'None' tag is equivalent to passing control to GRPC core until
+    # the deadline.
+    # TODO(atash) there's currently no way of getting a tag to pass to this
```

Comment:
As for misbehavior, it would only misbehave if core misbehaves; w.r.t. core, as long as pluck and next are not being called in an overlapping manner (which is ensured currently by the critical section), this shouldn't be a problem.